arch/x86/x86-hal/src/hal.rs

```diff
@@ -0,0 +1,71 @@
+//! System HAL implementation for x86 target.
+
+use std::convert::TryInto;
+use std::time::SystemTime;
+
+use aerugo_hal::system_hal::{SystemHal, SystemHardwareConfig};
+use bare_metal::CriticalSection;
+use once_cell::sync::Lazy;
+
+use crate::peripherals::Peripherals;
+
+/// Time when system was started
+static TIME_START: Lazy<SystemTime> = Lazy::new(SystemTime::now);
+
+/// HAL implementation for x86.
+pub struct Hal {
+    /// Hardware peripherals.
+    _peripherals: Peripherals,
+}
+
+impl Hal {
+    /// Frequency for the time types (TODO)
+    const TIMER_FREQ: u32 = 1_000_000;
+
+    /// Create new HAL instance.
+    pub const fn new(peripherals: Peripherals) -> Self {
+        Hal {
+            _peripherals: peripherals,
+        }
+    }
+}
+
+impl SystemHal for Hal {
+    type Instant = crate::time::TimerInstantU64<{ Hal::TIMER_FREQ }>;
+    type Duration = crate::time::TimerDurationU64<{ Hal::TIMER_FREQ }>;
+
+    fn configure_hardware(&mut self, _config: SystemHardwareConfig) {
+        todo!()
+    }
+
+    fn get_system_time(&self) -> Self::Instant {
+        Self::Instant::from_ticks(
+            TIME_START
+                .elapsed()
+                .expect("{}")
+                .as_nanos()
+                .try_into()
+                .unwrap(),
+        )
+    }
+
+    fn feed_watchdog(&mut self) {
+        todo!()
```

Comment:
Are we planning to implement watchdog on x86? We don't have HW, but we COULD have SW watchtog in another thread....

Reply:
But again, given that x86 on Linux is not real-time, does it make sense? We did have a watchdog for some Android SW, with period counted in minutes.

Reply:
I think that there is no point really, as we are only doing the x86 target for development purposes. I just added `todo` everywhere as a stub, but I can replace this one as we won't implement that at all.